.github/workflows/build-develop-unified.yml

```diff
@@ -0,0 +1,72 @@
+name: "Develop build (Unified)"
+
+on:
+  push:
+    branches:
+      - develop
+  merge_group:
+    types: [ checks_requested ]
+    branches: [ develop ]
+  pull_request:
+    branches:
+      - develop
+    types: [ opened, synchronize ]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.event.merge_group.head_sha }}
+  cancel-in-progress: true
+
+jobs:
+  # Build for PR and merge group validation (no store deployment)
+  build-validation:
+    if: github.event_name == 'pull_request' || github.event_name == 'merge_group'
+    uses: ./.github/workflows/build-unified.yml
+    with:
+      build-config: |
+        [
+          {
+            "flavor": "Dev",
+            "variant": "Debug",
+            "keystore-type": "debug",
+            "build-type": "apk",
+            "generate-version-file": false,
+            "deployment-targets": "[{\"type\": \"s3\"}]"
```

Comment:
`deployment-targets` being a JSON String within a JSON is a bit odd and makes maintenance a bit annoying. Can't it just be a JSON?

In order to pass it to the `deploy` job, we can do `toJSON(matrix.deployment-targets)`, so we can at least read the JSON object more easily.